### ISSUE TITLE
Deploy production Frontend and API

### ISSUE THREAD

**Issue by FoRVaiS** [Geek-Collectors-Network/geek-collectors-network#39]
### Description
We need to deploy our Frontend and API so that our stakeholders (GGR and Pat) will be able to view our progress at the end of each sprint.

### Solution
Use Vercel as web hosting platform for the Frontend and API.

There should be two deployments for both applications:
1. A bleeding edge build based on the most recent commit on the main branch.
2. A production release based on a published tag prefixed by `release-*` or a `production` tag, depending on what Vercel supports best.

**Comment by FoRVaiS** [comment]
Vercel does not support free GitHub Organization deployments.

Using AWS will provide plenty of services we can use for our app, which includes a virtual machine for deploying our frontend and api.

**Comment by FoRVaiS** [comment]
GitHub's student pack offers a $200 credit for 1 year of DigitalOcean (DO), although some calculation will have to be done to determine if using DO is the superior choice over AWS.

My Gut feeling says that seemingly free usage of select AWS services will dominate over DO's $200 credit offer, but again, price research will have to be done.

**Comment by FoRVaiS** [comment]
Earlier today, I started working on a pipeline to build and upload production docker images to the repo's ghcr.

The work was done in a fork as to not pollute the main branch's history with CI commits.

The functional workflow can be found [here](https://github.com/FoRVaiS/geek-collectors-network/blob/main/.github/workflows/production-deployment.yml). It still requires optimizations and some refactoring but minimally, it works.

After some thought, I took a look at Netlify, a Vercel competitor and it seems likely they don't paywall repositories from GitHub organizations. I'm going to apply some fixes to the frontend production build process then try to deploy our frontend and api to Netlify.
